gadopt/utility.py

```diff
@@ -302,6 +307,113 @@ def get_functionspace(mesh, h_family, h_degree, v_family=None, v_degree=None,
     return constructor(mesh, elt, **kwargs)
 
 
+class DiffusiveSmoothingSolver:
+    def __init__(
+            self,
+            function_space: fd.FunctionSpace,
+            wavelength: Number,
+            bcs: Optional[dict[int, dict[str, int | float]]] = None,
```

Comment:
`int | float` can be replaced by `Number`. I think I introduced the former as part of documentation changes, but Angus rightly pointed the existence of `Number`.